File: Chapter02/Exercise2_03/warc_schema_parser.py
from Chapter01.utilities01_py.helper_python import create_session
from Chapter02.utilities02_py.helper_python import sample_warc_loc, extract_raw_records, parse_raw_warc

if __name__ == "__main__":
    session = create_session(3, 'WARC Parser')
    session.sparkContext.setLogLevel('ERROR')  # avoids printing of info messages

    raw_records = extract_raw_records(sample_warc_loc, session)
    warc_records = raw_records.flatMap(lambda record: parse_raw_warc(record))

    warc_records.toDF().printSchema()
    print('Total # of records: ' + str(warc_records.count()))

File: Chapter02/Exercise2_03/WarcSchemaParser.scala
package Exercise2_03

import org.apache.hadoop.io.Text
import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SparkSession
import Utilities01.HelperScala.createSession
import Utilities02.HelperScala.{extractRawRecords, parseRawWarc, sampleWarcLoc}
import Utilities02.WarcRecord

/**
 * Code for parsing .warc files of the WARC corpus
 *
 * @author Phil, https://github.com/g1thubhub
 */
object WarcSchemaParser {

  def main(args: Array[String]) = {
    implicit val session: SparkSession = createSession(3, "WARC Parser")
    session.sparkContext.setLogLevel("ERROR") // avoids printing of info messages

    val rawRecords: RDD[Text] = extractRawRecords(sampleWarcLoc)
    val warcRecords: RDD[WarcRecord] = rawRecords
      .flatMap(parseRawWarc)

    import session.implicits._
    warcRecords.toDF().printSchema()
    println(s"Total # of records: ${warcRecords.count()}")

  }
}
